src/vm.rs

```diff
@@ -633,11 +633,10 @@ impl<T: GuestStagePageTable> Vm<T, VmStateFinalized> {
             counter_index_mask: u64,
             start_flags: PmuCounterStartFlags,
             initial_value: u64,
-            _active_cpu: &mut VmCpu,
+            active_cpu: &mut VmCpu,
         ) -> EcallResult<u64> {
             use sbi::api::pmu;
-            let pmu_info = PmuInfo::get();
-            let result = if pmu_info.get_counter_info(counter_index_base).is_some() {
+            let result = if active_cpu.is_counter_startable(counter_index_base) {
                 pmu::start_counters(
```

Comment:
I considered doing that, but how can we plumb the result of the ecall to the upper layer? Basically, we can't know whether the call will succeed or not, unless we want to assume that it will succeed based on some checks?